src/include/stir/recon_buildblock/RelativeDifferencePrior.h

```diff
@@ -193,6 +198,27 @@ class RelativeDifferencePrior:  public
   virtual bool post_processing();
  private:
   shared_ptr<DiscretisedDensity<3,elemT> > kappa_ptr;
+
+  //! The Hessian of the Relative Difference Prior
+  /*!
+   This function returns the hessian (second derivative) of the RDP.
+   It is assumed that the diagonal elements of the Hessian are 0, or the weighing is, and thus only compute the partial
+   derivative w.r.t \c x and \c y.
+   * @param x is the target voxel.
+   * @param y is the neighbourhood voxel.
+   * @param gamma is the edge preservation value controlling the transition between the quadratic and linear behaviour
+   * @param eps is a small non-negative value included to prevent division by zero, see epsilon.
+   * @return the second derivative of the Relative Difference Prior
+
+   */
+  static inline float Hessian(const float x, const float y, const float gamma, const float eps)
+  {
+    if (x > 0.0 || y > 0.0 || eps > 0.0)
```

Comment:
Just browsing, and I may be missing some context, so apologies if that is the case. 

Why should you check for `eps` here and return zero if negative? 

Would it not make more sense to have the `set_epsilon()` method validate that its positive (it has to be) and then assume here that `eps` is correct? This is a private static method, so no one else than this class can call it.

Even if that is not the case, if `eps` is an invalid value, why return `0.0` instead of error?

Also, perhaps a question for @KrisThielemans: why have **static private** methods here? optimization?

Reply:
`eps` is an term that was added to prevent divide by zeros and should be non-negative. However, it can be zero and therefore ignored. The check here is to return 0.0 if `x`, `y`, AND `eps` are zero. Otherwise, the function would compute `0^2/0^3`.

Reply:
makes sense!